### DISCUSSION TITLE
Best way to handle losing browser tab focus with deck.gl attribute transitions running

### DISCUSSION THREAD

**Discussion by JRVID** [visgl/deck.gl#7228] [Q&A] [answered]
Hey all,

I have a layer that renders live public transport vehicles (receiving updates every 30 seconds). I handle maintaining order currently since deck.gl uses index-based transitions (sad but manageable). I'm also aware that deck.gl purposely doesn't draw when the browser tab isn't focussed/active.

My question is: 

- Is it possible to actually force deck.gl to continue drawing even though browser tab isn't focussed?
- And if that's not possible is there some other solution to prevent all of my vehicles flying around the map like crazy if i blur the tab for a while and then come back to it?

Thanks!

**Comment by Pessimistress** [comment, marked as answer]
Deck.gl redraws on an animation loop by calling rAF. When you navigate away from the tab, it is the browser's decision to stop firing animation frames. With that said, when you come back to the tab the rendering should resume as usual. Can you open a bug so we can investigate?

**Reply by JRVID** [reply]
Thanks for the reply!

However apologies I may have worded this poorly. To attempt to clarify what I'm seeing:

When tab is initially focussed:

- Vehicle at position A gets an update from api -> layer is redrawn and animation moves vehicle to position B and then 30 seconds later another update and vehicle is animated from B to point C.

Then tab focus is lost:
- Vehicle receives update from point C -> point D from api - no redraw due to rAF not firing
- Leave for x amount of minutes and vehicle continues to get position updates from API until Vehicle gets an update to move to say point G

Refocus the tab:
- rAF stats again, layer redraws and animation is triggered and the vehicle (which was last at point C visually before we lost focus) then flies dramatically across the map directly from point C to point G - the latest position update.

Is this the expected behaviour? If not I'll raise a bug. If it is and there's no way to force rAF to continue firing even when tab focus is lost then i may need to come up with another workaround... potentially remove the animation until the next data change after tab focus event fires.

**Reply by Pessimistress** [reply]
That sounds like it's behaving as expected. It's usually favorable to watch the [visibilitychange event](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) and stop polling the server for updates when the page is hidden. When the user navigates back, you can request an accumulated update and apply it without any transition to avoid disorienting animations.

**Reply by JRVID** [reply]
Yep that makes sense. Thanks for the help! 😄
